miasm2/core/asmblock.py

```diff
@@ -696,66 +822,80 @@ def rebuild_edges(self):
         This method should be called if a block's '.bto' in nodes have been
         modified without notifying this instance to resynchronize edges.
         """
-        self._build_label2block()
-        for block in self._nodes:
+        for block in self.blocks:
             edges = []
             # Rebuild edges from bto
             for constraint in block.bto:
-                dst = self._label2block.get(constraint.label,
-                                            None)
+                dst = self._loc_key_to_block.get(constraint.label,
+                                                  None)
                 if dst is None:
                     # Missing destination, add to pendings
-                    self._pendings.setdefault(constraint.label,
-                                              set()).add(self.AsmCFGPending(block,
-                                                                            constraint.c_t))
+                    self._pendings.setdefault(
+                        constraint.label,
+                        set()
+                    ).add(
+                        self.AsmCFGPending(
+                            block,
+                            constraint.c_t
+                        )
+                    )
                     continue
-                edge = (block, dst)
+                edge = (block.label, dst.label)
                 edges.append(edge)
                 if edge in self._edges:
                     # Already known edge, constraint may have changed
                     self.edges2constraint[edge] = constraint.c_t
                 else:
                     # An edge is missing
-                    self.add_edge(edge[0], edge[1], constraint.c_t)
+                    self.add_constraint(edge[0], edge[1], constraint.c_t)
 
             # Remove useless edges
-            for succ in self.successors(block):
-                edge = (block, succ)
+            for succ in self.successors(block.label):
+                edge = (block.label, succ)
                 if edge not in edges:
                     self.del_edge(*edge)
 
     def get_bad_blocks(self):
         """Iterator on AsmBlockBad elements"""
         # A bad asm block is always a leaf
-        for block in self.leaves():
+        for label in self.leaves():
+            block = self._loc_key_to_block.get(label, None)
```

Comment:
If block is None, it should also returns an `AsmBlockBad` (this is a transient situation, we have both blocks with AsmBlockBad, and loc_key without corresponding block, to represent the same thing...)

Reply:
done